lib/api/core.dart

```diff
@@ -37,16 +37,20 @@ abstract class ApiConnection {
   Future<String> post(String route, Map<String, dynamic>? params);
 }
 
+MapEntry<String, String> authHeader(Auth auth) {
+  // TODO memoize
+  final authBytes = utf8.encode("${auth.email}:${auth.apiKey}");
+  return MapEntry('Authorization', 'Basic ${base64.encode(authBytes)}');
+}
+
 /// An [ApiConnection] that makes real network requests to a real server.
 class LiveApiConnection extends ApiConnection {
   LiveApiConnection({required super.auth});
 
   Map<String, String> _headers() {
-    // TODO memoize
```

Comment:
This comment wasn't super clear, but I think the memoization I would ideally want is of the entire `Map`. I think that means:
* Let's have the helper function encapsulate returning the entire Map, not the individual header. (If the different use sites later diverge in what they need, we can refactor then.)
* Maybe put the comment at the outside of that function rather than the inside, to be a bit clearer about that than this comment was :-)